### ISSUE TITLE
[AL+Parallel] Conker's Bad Fur Day (FPS issues)

### ISSUE THREAD

**Issue by HelicopterP** [libretro/mupen64plus-libretro-nx#157]
I'm runnig retroarch on windows 10 and core i7 8700 (non k).
Parallel RSP + angrylion (6 threads, filtered VI)
During intro scene when Conker enters the bar I get huge FPS drops to high 39s - low 40s, none of my 6 real cores get 100% utilisation. 
The weired thing is if I press space on keyboard (fast forward mode) I can almost get fullspeed during same scenes (something like 58-65 fps before Conker opens the door and 49-75 after), core1 gets 100% utilisation.

Gliden64 is completely unaffected (fullspeed nomatter what).

**Comment by m4xw** [comment]
Fast forward just ignores vsync and audio sync, the fps counter probably doesn't have enough time to synchronize.
Also 100% utilisation doesnt mean much if the bottleneck for example is memory or throttling related.
I will give this some investigation but I am likely just gonna close the Issue.
Do you use hyperthreading?

**Comment by HelicopterP** [comment]
-> the fps counter probably doesn't have enough time to synchronize.
This isn't a fake FPS increase definitely. I start the game with framelimiter enabled and wait for that scene. The game runs slow, audio is garbled (which is obviously expected below 60fps), then i press fast forward key and the game immediately runs faster and audio is much better. It's not about frame counter - I can see it and feel it.

-> Fast forward just ignores vsync and audio sync
But how is Gliden64 unaffected? If it was a syncing issue, wouldn't I got slowdown with any graphics plugin?

->Also 100% utilisation doesnt mean much if the bottleneck for example is memory or throttling related.
It's definatelly not related to throttling - CPU clock stays the same during my tests.  If it's memory related, then It should be slow nomatter waht, right? But it's not.

-> Do you use hyperthreading?
Yes, I do, but I don't think it's related. Actually I get lower overall performance with max threads settings vs just 6 threads. So, default setting might be unoptimal in my case (or maybe on all CPUs with HT and SMT), but It's not about this issue.

**Comment by m4xw** [comment]
> But how is Gliden64 unaffected? If it was a syncing issue, wouldn't I got slowdown with any graphics plugin?

GLideN doesnt hammer your CPU caches

> It's definatelly not related to throttling - CPU clock stays the same during my tests. If it's memory related, then It should be slow nomatter what, right? But it's not.

See above

**Comment by HelicopterP** [comment]
-> Yeah, that explains why Glide is faster, but it doesn't explain why bottleneck only manifests itself with frame limiter enabled.
I'm not making this up - for me enabling framelimiter during intro slows things down. It's literally like:
slow - press space and its fast - press space again and it's slow - press space one more time and it's fast again.

**Comment by m4xw** [comment]
I am not assuming you make this up

**Comment by Testsr** [comment]
Fast forward also helps Mario 64,Banjo Kazooie, as well.

**Comment by HelicopterP** [comment]
Ok, I've done some additional testing and found out that this issue only affects Vulkan renderer. OpenGl renderer runs  fine with and without fps limiter.
There should be no differnece because angrylion is a software renderer, but yet this bug exists. I have no idea whether it's a muppen or retroarch bug.
Tests were done on GTX1070  on Windows 10 with latest drivers.

**Comment by m4xw** [comment]
@HelicopterP Good info, thanks.

**Comment by m4xw** [comment]
Might be resolved with recent mupen core changes, please confirm https://git.m4xw.net/libretro/cores/mupen64plus-libretro-nx/pipelines/4974/builds

**Comment by HelicopterP** [comment]
Ok, I'm sorry for not getting down to the root of this issue earlier, but I finally figured it out.
Your last commit doesn't fix this issue, but I don't think you can do anything about it sadly.
So this is what i figured out:
Open GL+Vsync (in retroarch settings) + Gsync - good framerate, no screen tearing.
Open GL+Vsync (in retroarch settings) - low framerate, no screen tearing.
OpenGL without Vsync - good framerate, screen tearing.
Vulkan without Vsync - good framerate, screen tearing.
So, all of the above is perfectly normal and expected.

But combination of Vulkan, Vsync and Gsync totaly ruins the framerate, it tanks way bellow of what it should be (i.e. 65 fps in FF-mode vs 33 fps in normal mode). Pressing space key disables Vsync and emulator runs much faster. So the real problem is in combination of this three: Vulkan, Vsync and Gsync - change any one of the variables and the issue is gone.

I'm not sure about the best way to handle it now. Should I open an Issue on Retroarch issue tracker?

**Comment by m4xw** [comment]
I still have one thing to try (wait till its done), https://git.m4xw.net/libretro/cores/mupen64plus-libretro-nx/pipelines/4976/builds

**Comment by HelicopterP** [comment]
No difference, sadly. Vulkan and Gsync just don't play nice unfortunatelly.

**Comment by HelicopterP** [comment]
I did some tests on Vulkan 1.2 Beta driver.  Apparently Vsync implementation is bugged in current stable driver (Vsync simply doesn't work in exclusive fullscreen mode), but works as it should in beta driver. Unfortunately there is something seriously wrong with Vsync and Muppen in Retroarch. All test were made in exclusive fullscreen mode on Vulkan retroarch driver.

Enabling Vsync tanks performance. For example I get drops to 40 fps during intro animation with Vsync; up to 160 FPS with fast forwarding; and stable 60 FPS without Vsync and FF. The weired thing is there is absolutely no screen tearing in Vulkan mode with Vsync disabled, so right now Vsync is doing nothing but killing performance.
The same issue affects other N64 games by the way. It is also present in ParaLLEl.

I wanted to test your builds again with current beta driver as I suspect previous reports might be incorrect due to nVidia bug, but those builds were removed apparently.
